src/mono/wasm/runtime/net6-legacy/method-calls.ts

```diff
@@ -337,18 +338,26 @@ export function mono_wasm_change_case(exceptionMessage: Int32Ptr, culture: MonoS
     }
 }
 
-function get_uft16_string (ptr: number, length: number): string{
-    const view = new Uint16Array(Module.HEAPU16.buffer, ptr, length);
-    let string = "";
-    for (let i = 0; i < length; i++)
-        string += String.fromCharCode(view[i]);
-    return string;
+export function mono_wasm_compare_string(exceptionMessage: Int32Ptr, culture: MonoStringRef, str1: number, str1Length: number, str2: number, str2Length: number, options: number) : number{
+    const cultureRoot = mono_wasm_new_external_root<MonoString>(culture);
+    try{
+        const cultureName = conv_string_root(cultureRoot);
+        const string1  = get_uft16_string(str1, str1Length);
+        const string2 = get_uft16_string(str2, str2Length);
+        const casePicker = (options & 0x1f);
+        const locale = (cultureName && cultureName?.trim()) ? cultureName : undefined;
```

Comment:
this logic seems a little weird to me, if we're trimming the name shouldn't we be using the trimmed name here? or is this just meant to be IsNullOrWhiteSpace

Reply:
It was meant to be IsNullOrWhitespace but now I think doing `if (cultureName)` check will do good as well. I will simplify it